misc/python/materialize/cli/mzcompose.py

```diff
@@ -611,7 +611,23 @@ def handle_composition(
 BuildCommand = DockerComposeCommand("build", "build or rebuild services")
 ConfigCommand = DockerComposeCommand("config", "validate and view the Compose file")
 CreateCommand = DockerComposeCommand("create", "create services", runs_containers=True)
-DownCommand = DockerComposeCommand("down", "stop and remove resources")
+
+
+class DownCommand(DockerComposeCommand):
+    def __init__(self) -> None:
+        super().__init__(
+            "down",
+            "clean up containers from the composition",
+        )
+
+    def run(self, args: argparse.Namespace) -> Any:
+        # --remove-orphans needs to be in effect at all times otherwise
+        # services added to a composition after the fact will not be cleaned up
+        args.unknown_subargs.append("--remove-orphans")
+        setattr(args, "help", False)
```

Comment:
This `setattr` seems sketchy! Is it necessary?

Reply:
Yes, there are `if args.help:` checks downstream in the code that will throw unless this is set. There are already several uses of `setattr` in the code, including against `help` in the `run` method of `RunCommand` .

Reply:
> There are already several uses of setattr in the code, including against help in the run method of RunCommand .

Those are necessary because `RunCommand` overrides the `def configure` method, which is what should cause `args.help` to get set to either `True` or `False`. You're not overriding `configure` here, so it's surprising that you need to set this yourself! Makes me think that something is going wrong.

Reply:
I removed it without adverse consequences. Thanks!